### PR TITLE
cmake, rgw: always enable Boost.context

### PR DESCRIPTION
**Pull request by adamemerson**
As Boost.Context supports s390 as of 1.72, bump Boost dependency to that version and remove conditional compilation around coroutines.

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@adamemerson yay

File: CMakeLists.txt [hunk 1]
```diff
@@ -593,7 +593,7 @@ if(WITH_SYSTEM_BOOST)
   if(BOOST_ROOT AND CMAKE_LIBRARY_ARCHITECTURE)
     set(BOOST_LIBRARYDIR "${BOOST_ROOT}/lib/${CMAKE_LIBRARY_ARCHITECTURE}")
   endif()
-  find_package(Boost 1.67 COMPONENTS ${BOOST_COMPONENTS} REQUIRED)
+  find_package(Boost 1.72 COMPONENTS ${BOOST_COMPONENTS} REQUIRED)
```
Comment (tchaikov) [inline CMakeLists.txt:584]: could you s/build/cmake/ in the title of this commit?

File: ceph.spec.in [hunk 2]
```diff
@@ -1173,11 +1173,6 @@ ${CMAKE} .. \
 %if 0%{with ocf}
     -DWITH_OCF=ON \
 %endif
-%ifarch aarch64 armv7hl mips mipsel ppc ppc64 ppc64le %{ix86} x86_64 s390x
```
Comment (tchaikov) [inline ceph.spec.in:1176]: why? this matches with https://www.boost.org/doc/libs/1_72_0/libs/context/doc/html/context/architectures.html .

**Review by tchaikov — CHANGES_REQUESTED** [review]
please s/build/cmake/ in the title

also might want to move the `ifarch` change into a separate commit. as it's not related to s390x support.

File: CMakeLists.txt (hunk 1 above)
Comment (adamemerson) [reply CMakeLists.txt:584]: Done

File: ceph.spec.in (hunk 2 above)
Comment (adamemerson) [reply ceph.spec.in:1176]: Because we don't have a WITH_BOOST_CONTEXT option any more. Though I'll break it out into a separate commit.

Comment (tchaikov) [reply ceph.spec.in:1176]: oic. thanks!

**Review by tchaikov — APPROVED** [review]

File: src/tools/CMakeLists.txt [hunk 3]
```diff
@@ -16,7 +16,6 @@ else()
 endif()
 install(TARGETS rados DESTINATION bin)
 
-if(WITH_BOOST_CONTEXT AND NOT WIN32)
```
Comment (cbodley) [inline src/tools/CMakeLists.txt:19]: any idea why the WIN32 part was there?

**Review by cbodley — COMMENTED** [review]
thanks @adamemerson, s390x users everywhere will rejoice!

File: src/tools/CMakeLists.txt (hunk 3 above)
Comment (adamemerson) [reply src/tools/CMakeLists.txt:19]: Some people 'stubbed' it doing a windows port, but I don't know why. I put that back.

**Comment by adamemerson** [comment]
https://pulpito.ceph.com/aemerson-2020-10-23_18:51:18-rgw-wip-context4all-distro-basic-smithi/

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-10-23_10:21:03-rados-wip-kefu-testing-2020-10-23-1631-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-10-24_10:41:09-rados-wip-kefu-testing-2020-10-24-1306-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-10-24_16:10:41-rados-wip-kefu-testing-2020-10-24-1306-distro-basic-smithi/

failures tracked by

- https://tracker.ceph.com/issues/46323

**Comment by adamemerson** [comment]
Anything more this needs? We've got RADOS and RGW runs.

**Pull request merged**